src/java.desktop/share/classes/javax/swing/text/rtf/RTFParser.java

```diff
@@ -360,4 +371,34 @@ public void close()
     super.close();
   }
 
+    private char[] ca = new char[1];
+    private CharBuffer decoderCB = CharBuffer.wrap(ca);
+
```

Comment:
So I know in theory what needs to be done here, and the test looks thorough,
but basically you've dumped code on us with zero explanation of the fix,
and I've spent quite a while figuring out some of it but I think you owe it to reviewers to do a much better job
of explaining the logic and flow. 
The function below needs particular explanation. 
One thing (not the only thing !) I want you to describe is what you are using \uFFFD for in various places in this change.